### PR TITLE
Lessons model 

### PR DESCRIPTION
**Pull request by mfroment**
(on behalf of @pillowbug and @jy-ko)
Lesson/User/Booking models, controller, routes initial implementation

### CODE REVIEW

File: app/controllers/bookings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class BookingsController < ApplicationController
+  def create
+    @booking = Booking.new
+    @user = current_user
+    @lesson = Lesson.find(params[:lesson_id])
+    @booking.lesson = @lesson
+    @booking.user = @user
+    @booking.save!
```
Comment (mfroment) [inline app/controllers/bookings_controller.rb]: Unneeded?

File: app/controllers/bookings_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    if current_user
+      @booking.save!
```
Comment (mfroment) [inline app/controllers/bookings_controller.rb]: Should replace with test on successfully saving ; if not, re-render page.
Possible cases = already booked, conflict with other booked lesson (--> need extra validations)

File: app/controllers/bookings_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      redirect_to user_path(@user)
+    else
+      redirect_to new_user_session_path
```
Comment (mfroment) [inline app/controllers/bookings_controller.rb]: OK for now, possibly improve by providing a way to return to lesson page after login.

File: app/controllers/pages_controller.rb [hunk 4]
```diff
@@ -1,6 +1,8 @@
 class PagesController < ApplicationController
-  skip_before_action :authenticate_user!, only: [:home]
-
+  skip_before_action :authenticate_user!, only: [:home, :lessons]
   def home
   end
+
+  def lessons
```
Comment (mfroment) [inline app/controllers/pages_controller.rb]: Is this needed?

File: app/controllers/bookings_controller.rb (hunk 1 above)
Comment (jy-ko) [reply app/controllers/bookings_controller.rb]: yes

**Pull request merged**